netkan/netkan/metadata.py

```diff
@@ -0,0 +1,65 @@
+import json
+import re
+from pathlib import Path
+from hashlib import md5
+
+
+class Netkan:
+
+    kref_pattern = re.compile('^#/ckan/([^/]+)/(.+)$')
+
+    def __init__(self, filename=None, contents=None):
+        if filename:
+            self.filename = Path(filename)
+            self.contents = self.filename.read_text()
+        else:
+            self.contents = contents
+        self._raw = json.loads(self.contents)
+
+    def __getattr__(self, name):
+        # Extract kind + mod_id from the kref
+        if name in ['kind', 'mod_id']:
```

Comment:
So it turns out that `kind` is a valid defined metadata property outside of a kref context (it distinguishes packages from metapackages, and is used in the mirroring), so we'll need another name for that first part of the kref to avoid conflicting.

https://github.com/KSP-CKAN/CKAN/blob/master/Spec.md#kind

`Netkan.Model.RemoteRef` calls it `Source`. Maybe we could call it `kref_source` here?

Reply:
Yeah, that's a point. I also don't like `mod_id` as it's kinda confusing. I'll refactor both.